src/gui/sharee.h

```diff
@@ -35,35 +35,47 @@ Q_DECLARE_LOGGING_CATEGORY(lcSharing)
 
 class Sharee
 {
+    Q_GADGET
+    Q_PROPERTY(QString format READ format)
+    Q_PROPERTY(QString shareWith MEMBER _shareWith)
+    Q_PROPERTY(QString displayName MEMBER _displayName)
+    Q_PROPERTY(QString iconUrlColoured MEMBER _iconUrlColoured)
+    Q_PROPERTY(Type type MEMBER _type)
+
 public:
     // Keep in sync with Share::ShareType
-    enum Type {
-        User = 0,
-        Group = 1,
-        Email = 4,
-        Federated = 6,
-        Circle = 7,
-        Room = 10
-    };
+    enum Type { Invalid = -1, User = 0, Group = 1, Email = 4, Federated = 6, Circle = 7, Room = 10, LookupServerSearch = 999, LookupServerSearchResults = 1000 };
+    Q_ENUM(Type);
+    explicit Sharee() = default;
+    explicit Sharee(const QString &shareWith, const QString &displayName, const Type type, const QString &iconUrl = {});
 
-    explicit Sharee(const QString shareWith,
-        const QString displayName,
-        const Type type);
+    Q_REQUIRED_RESULT QString format() const;
+    Q_REQUIRED_RESULT QString shareWith() const;
+    Q_REQUIRED_RESULT QString displayName() const;
+    Q_REQUIRED_RESULT QString iconUrl() const;
+    Q_REQUIRED_RESULT QString iconUrlColoured() const;
+    Q_REQUIRED_RESULT Type type() const;
```

Comment:
We should avoid using `Q_REQUIRED_RESULT` as this is not documented in Qt and seems to be a private thing -- better use `[[nodiscard]]`